docs/userguides/compile.md

```diff
@@ -110,3 +110,26 @@ vyper.compile([Path("path/to/contract.vy")])
 solidity = compilers.get_compiler("solidity", settings=settings["solidity"])
 vyper.compile([Path("path/to/contract.sol")])
 ```
+
+## Compile Source Code
+
+Instead of compiling project source files, you can compile code (str) directly:
+
+```python
+from ape import accounts, compilers
+
+CODE = """
+   ... source code here
+"""
+
+container = compilers.compile_source(
+   "vyper",
+   CODE,
+   settings={"vyper": {"version": "0.3.7"}}, 
+   contract_type_overrides={"contractName": "MyContract"}
```

Comment:
yes it does, this is the manager.

Reply:
i didnt do kwargs on the manager, just the api, because there are 2 dicts here and it seems confusing which one is **kwargs and which one isnt

Reply:
plus the manager doesnt even return a contract type - it returns a container, so the context is even more removed when using the **kwargs method at the manager level, so it only at api level

Reply:
here is the manager signature:

```
 def compile_source(
        self,
        compiler_name: str,
        code: str,
        settings: Optional[Dict] = None,
        contract_type_overrides: Optional[Dict] = None,
    ) -> ContractContainer:
```

here is api signature:

```python
    def compile_code(
        self,
        code: str,
        base_path: Optional[Path] = None,
        **contract_type_overrides,
    ) -> ContractType:
```

I think i'd like avoiding ** altogether but idc.
as long as i can pass both settings and overrides separately, the use-case is covered.

Reply:
yeah, that does seem kind of confusing altogether

Reply:
how so?

Reply:
tell me the signatures you prefer.

Reply:
the `**` based part is what it makes confusing imo, it is much clearer when using dictionaries.

Reply:
the kwargs in the API interface will allow it to change a bit more easily over time.

I think doing the same in the manager interface to match would be helpful. the settings override being a dict I don't think is that confusing if the contract type overrides are kwargs. in fact, might be less confusing

Reply:
I am going to change it back to that, let me know if you have any other ideas though 👍🏻

Reply:
kwargs causes name clashing between contract type overrides and actual method kwargs. if anything it makes change harder because now if we attributes to ContractType, we have to make sure they are not clashing with kwarguments not part of the overrides.

Reply:
I anticipate problems with this... but ok...